Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp

```diff
@@ -0,0 +1,1011 @@
+        continue;
+    }
+
+    if (labels & AtomIndexLabels) {
+      if (setLabel(atom, indexOffset - atom->getIdx(), foundLabels, maxLabel,
+                   relabel, "IndexLabels"))
+        nextOffset++;
+        continue;
+    }
+  }
+
+  indexOffset -= nextOffset;
+
+  std::map<int, int> labels;
+  for (RWMol::AtomIterator atIt = core.beginAtoms(); atIt != core.endAtoms();
+       atIt++) {
+    Atom *atom = *atIt;
+    int rlabel;
+    if(atom->getPropIfPresent(RLABEL, rlabel)) {
+      labels[atom->getIdx()] = rlabel;
+    }
+  }
+
+  MolOps::AdjustQueryParameters adjustParams;
+  adjustParams.makeDummiesQueries = true;
+  adjustParams.adjustDegree = onlyMatchAtRGroups ?
+      MolOps::AdjustDegree::HeavyDegree : MolOps::AdjustDegree::NoAdjust;
+  adjustQueryProperties(core, &adjustParams);
+  
+  for(std::map<int,int>::iterator it=labels.begin(); it!=labels.end(); ++it)
+    core.getAtomWithIdx(it->first)->setProp(RLABEL, it->second);
+
+  return true;
+}
+
+namespace {
+// RGroup Class to hold the attached bits
+struct RGroupData {
+  boost::shared_ptr<RWMol> combinedMol;
+  std::vector<boost::shared_ptr<ROMol> > mols;  // All the mols in the rgroup
+  std::set<std::string> smilesSet;              // used for rgroup equivalence
+  std::string smiles;  // smiles for all the mols in the rgroup (with attachments)
+  std::set<int> attachments;                    // attachment points
+  
+  void add(boost::shared_ptr<ROMol> newMol,
+           const std::vector<int> &rlabel_attachments) {
+    // some fragments can be add multiple times if they are cyclic
+    for (size_t i=0; i<mols.size(); ++i) {
+      if (newMol.get() == mols[i].get())
+        return;
+    }
+
+    std::copy( rlabel_attachments.begin(), rlabel_attachments.end(),
+               std::inserter( attachments, attachments.end() ) );
+
+    mols.push_back(newMol);
+    std::string smi = MolToSmiles(*newMol, true);
+    smilesSet.insert(smi);
+    if (!combinedMol.get()) {
+      combinedMol = boost::shared_ptr<RWMol>(new RWMol(*mols[0].get()));
+    } else {
+      ROMol *m = combineMols(*combinedMol.get(), *newMol.get());
+      combinedMol.reset(new RWMol(*m));
+      delete m;
+    }
+    smiles = getSmiles();
+  }
+
+  std::map<int,int> getNumBondsToRlabels() const {
+    std::map<int,int> rlabelsUsedCount;
+
+    for (ROMol::AtomIterator atIt = combinedMol->beginAtoms();
+         atIt != combinedMol->endAtoms();
+         ++atIt) {
+      Atom *atom = *atIt;
+      int rlabel; 
+      if(atom->getPropIfPresent<int>(RLABEL, rlabel))
+        rlabelsUsedCount[rlabel] += 1;
+    }
+    return rlabelsUsedCount;
+  }
+  
+  bool isHydrogen() const {  // is the rgroup all Hs
+    for(size_t i=0; i<mols.size(); ++i ) {
+      for (ROMol::AtomIterator atIt = mols[i]->beginAtoms();
+           atIt != mols[i]->endAtoms(); ++atIt) {
+        if ((*atIt)->getAtomicNum() > 1) return false;
+      }
+    }
+    return true;
+  }
+
+ private:
+  std::string getSmiles()
```

Comment:
If I recall correctly this is intentional to "fix" ordering around an atom in the case of stereo matches.  It may be I'm thinking too much about it though.

I should add the "."